app/api/channel_api.py

```diff
@@ -0,0 +1,233 @@
+
+    Args:
+        id (str): [channel id which is being edited]
+
+    Raises:
+        HTTPException: ["Oops! Unexpected mongodb error occurred" or
+                        "channel Not Found"]
+
+    Returns:
+        Channel [dict]: [updated channel info]
+    """
+    try:
+        response = update_channel(id, new_data)
+        return response
+    except returnExceptions as err:
+        raise HTTPException(
+            status_code=Code.error_enum_http[err.error_code], detail=str(err)
+        )
+
+
+@channels_router.post("/channels/join/{id}")
+async def add_user_to_channel(id: str, user_data: dict):
+    """
+    Endpoint to add user to a channel
+
+    Args:
+        id (str): [channel_id]
+        user_data (dict): [user_id[str]: [user being added],
+                            req_user[str]: [user sending req(may or may not be admin)]]
+
+    Raises:
+        HTTPException: ["Oops! Unexpected mongodb error occurred" or
+                        "channel Not Found" or
+                        "User Already Exists" or
+                        "request denied, user doesn't have permission for this request"
+                        "channel type is undefined"]
+
+    Returns:
+        membership[dict]: [membership of user is returned]
+    """
+    try:
+        response = join_user(id, user_data)
+        return response
+    except returnExceptions as err:
+        raise HTTPException(
+            status_code=Code.error_enum_http[err.error_code], detail=str(err)
+        )
+
+
+@channels_router.post("/channels/leave/{id}")
+async def leave_channel(id: str, user_data: dict):
+    """
+    Endpoint to for an user to leave a channel
+
+    Args:
+        id (str): [channel_id]
+        user_data (dict): [user_id[str]: [user being added],
+                            req_user[str]: [user sending req(may or may not be admin)]]
+    Raises:
+        HTTPException: ["Oops! Unexpected mongodb error occurred" or
+                        "channel Not Found" or
+                        "User Not Found" or
+                        "user not in channel"]
+
+    Returns:
+        [type]: [description]
+    """
+    try:
+        response = user_leave(id, user_data)
+        return response
+    except returnExceptions as err:
+        raise HTTPException(
+            status_code=Code.error_enum_http[err.error_code], detail=str(err)
+        )
+
+
+@channels_router.delete("/channels/delete/{id}", response_description="Delete channel")
+async def delete_channel(id: str, user_data: dict):
+    """
+    Endpoint to delete a channel
+
+    Args:
+        id[str]: [channel_id]
+        user_data[dict]: [user_id[str]: [user being added],
+                            req_user[str]: [user sending req(may or may not be owner)]]
+    Raises:
+        HTTPException: ["Oops! Unexpected mongodb error occurred", or
+                        "channel Not Found" or
+                        "request denied, user doesn't have permission for this request"]
```

Comment:
1. remove the square brackets in the description
2. this doesnt explain much. What does each message mean? when is it raised? Plus, don't put the error messages you will send as docs